lib/Doctrine/ORM/Mapping/Driver/AnnotationDriver.php

```diff
@@ -472,7 +472,7 @@ public function loadMetadataForClass($className, ClassMetadata $metadata)
 
                 // Check for `fetch`
                 if ($associationOverride->fetch) {
-                    $override['fetch'] = constant('Doctrine\ORM\Mapping\ClassMetadata::FETCH_' . $associationOverride->fetch);
+                    $override['fetch'] = constant(ClassMetadata::class . '::FETCH_' . $associationOverride->fetch);
```

Comment:
This won't work since `ClassMetadata` points to `Doctrine\Common\Persistence\Mapping\ClassMetadata` and it doesn't have the constants.